homework/Kiet_Tran/lesson_1/happy_number.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+class Solution {
+
+    /**
+     * @param Integer $n
+     * @return Boolean
+     */
+    function isHappy(int $n): bool {
+        if ($n == 1) {
+            return true;
+        }
+
+        $appeared = [];
```

Comment:
this costs O(n) space. Can use Floy's algorithm to improve to O(1) space (similar to linked_list_cycle)

Reply:
Floy's algorithm relies on a virtual linked_list_cycle
We don't need to use more array to store steps, so space is O(1)
I updated, thank you